Code/Editor/EditorFramework/IPC/EngineProcessConnection.cpp

```diff
@@ -392,10 +390,23 @@ ezResult ezEditorEngineProcessConnection::RestartProcess()
 
   ezLog::Dev("Transmitting open documents to Engine Process");
 
-  // resend all open documents
+  ezHybridArray<ezAssetDocument*, 6> docs;
+  docs.Reserve(m_DocumentByGuid.GetCount());
+
+  // Resend all open documents. Make sure to send main documents before child documents.
   for (auto it = m_DocumentByGuid.GetIterator(); it.IsValid(); ++it)
   {
-    SendDocumentOpenMessage(it.Value(), true);
+    docs.PushBack(it.Value());
+  }
+  docs.Sort([](const ezAssetDocument* a, const ezAssetDocument* b) {
+    if (a->IsMainDocument() != b->IsMainDocument())
+      return a->IsMainDocument() > b->IsMainDocument();

```

Comment:
You don't need this bool-as-int-and-then-compare behavior, because you already know that a != b. Can just return whether a is the main document.